### PR TITLE
Add heartbeatUntilPromise to make it easy to keep a job lock.

### PR DESCRIPTION
**Pull request by dlecocq**
@evanbattaglia @b4hand - I was sitting down to write this bit of functionality for my own project, but thought it'd be useful here, too.

This introduces `job.heartbeatUntilPromise`, which allows users to provide a promise and until that promise resolves, the job will heartbeat as needed.

```javascript
class MyJobClass {
  static queue(job) {
    // Promise that describes the real work of my job
    const realWork = ...;

    // Even if the real work takes a long time, we don't lose our lock
    return job.heartbeatUntilPromise(realWork)
      .then(() => job.complete());
  }
}
```

### CODE REVIEW

**Comment by dlecocq** [comment]
This one should also be accompanied by a version bump after #13 and #14

File: lib/job.js [hunk 1]
```diff
@@ -206,6 +206,34 @@ class Job {
...
+   * Before the job lock is lost, the heartbeat is run. The padding controls how long
+   * (in seconds) before the job lock loss to run the heartbeat. The default value is
+   * 10, meaning that the heartbeat will run 10 seconds before the lock is set to
+   * expire.
+   */
+  heartbeatUntilPromise(promise, padding) {
+    // The promise is wrapped in Bluebird to ensure it adheres to the expected API
+    const wrappedPromise = Promise.resolve(promise);
+    // Default to 10 seconds
+    padding = padding || 10;
```
Comment (evanbattaglia) [inline lib/job.js]: You may be able to use default parameters (I'm not 100% sure at which node version this was supported, I'm on node 6.10)

```
$ node
> function x(foo = 5) { console.log(foo); }
undefined
> x()
5
undefined
```

Comment (dlecocq) [inline lib/job.js]: Yeah, I tried that originally, but currently the build matrix supports node 4, which does not support it. I figure it's such a minor transformation that we may as well do the `||` style for the sake of supporting an older version.

Comment (evanbattaglia) [inline lib/job.js]: Cool, it's nice to have a build matrix that shows what works in what version.

**Comment by dlecocq** [comment]
@evanbattaglia I just pushed an update that bumps the version on this branch to `0.1.2`

**Pull request merged**